### PR TITLE
Phase 2: Implement request batching for shopper tracking events

### LINKED ISSUE #9677: Update shopper Tracks events enablement criteria
### Describe the issue

Shopper usage tracking utilizes Ajax requests to record activity at various touchpoints during the checkout process. Each record event emits a separate Ajax request. For high-traffic stores, there can be a high amount of Tracks originated requests hitting the merchant server. 

Shopper Tracks events are enabled by default and does not follow the merchant's Tracks opt-out checkbox.

1. We need to investigate ways to reduce the number of requests going to the merchant server. One option is making these requests in batches. One thing to be cautious of with such an approach is that we need to store the correct timestamp of the user action (not the request timestamp). This will require some refactoring of [tracks_build_event_obj](https://github.com/Automattic/woocommerce-payments/blob/598dc89c3b2b95371c5ea38a58bbcb968c817588/includes/class-woopay-tracker.php#L359) function.

2. Consider introducing a new filter to disable front-end tracking altogether when needed.
3. Consider respecting the existing merchant's Tracks opt-out checkbox.

### PR DESCRIPTION
**Pull request by rtio**
Fixes #9677

**Note:** This PR builds on top of Phase 1 (#11188) and should be merged after it.

#### Changes proposed in this Pull Request

This PR implements request batching for shopper tracking events to dramatically reduce server load on high-traffic stores. This is Phase 2 of the solution for WOOPMNT-4451.

**Problem:** 
Each shopper tracking event generates a separate Ajax request to `/wp-admin/admin-ajax.php`. For high-traffic stores, this creates significant server load:
- 5-10+ separate requests per checkout session
- Each request loads the full WordPress admin context
- During promotions, this compounds exponentially
- Customer reported this as a major issue during high-traffic periods

**Solution:**

**Client-side batching:**
- Queue tracking events instead of sending immediately
- Debounced flush after 2 seconds of inactivity
- Batch multiple events into single request
- Immediate flush on page unload to prevent data loss
- Preserve accurate event timestamps from client

**Server-side batch handler:**
- New `platform_tracks_batch` Ajax action
- Process multiple events in single handler
- Support client-provided timestamps (no data loss)
- Efficient validation and processing

**Express checkout improvements:**
- Add tracking enablement check for Apple/Google Pay events
- Respect `wcpaySettings.enableShopperTracking` flag
- Prevent hardcoded tracking when opt-out is enabled

**Why this approach:**
- Minimal code changes, leverages existing infrastructure
- Debouncing groups rapid events naturally (button clicks, page views)
- Page unload handler ensures no events are lost
- Client timestamps preserve accurate event timing
- Backward compatible (falls back to individual requests if needed)

**Trade-offs:**
- 2-second delay before events are sent (acceptable for analytics)
- Slightly more complex client-side code
- Additional server endpoint to maintain

**How this code can break:**
- Page unload handler might not fire in all browsers (rare edge case)
- Debounce timing could affect real-time dashboards
- Batched requests could timeout if queue gets too large

**What we're doing to prevent breakage:**
- Comprehensive unit tests for batch handler
- Error handling for failed batch requests
- beforeunload handler for reliable flushing
- Timestamps preserve event accuracy
- Existing single-event endpoint still works

#### Testing instructions

**Test 1: Verify batching reduces requests**
1. Open browser DevTools → Network tab
2. Filter for `admin-ajax.php`
3. Visit product page, add to cart, go to checkout (multiple tracking events)
4. Wait 2 seconds after last action
5. Verify you see:
   - ONE request to `action=platform_tracks_batch` instead of 5+ separate requests
   - Request payload contains multiple events with timestamps

**Test 2: Verify events are still tracked correctly**
1. Complete a checkout with batching enabled
2. Verify all expected events are recorded (product view, cart view, checkout view, etc.)
3. Check that event timestamps are accurate (not all the same server time)

**Test 3: Test page unload flushing**
1. Visit product page (triggers event)
2. Immediately navigate away (before 2-second debounce)
3. Check network tab - event should be sent immediately on navigation
4. Verify no events are lost

**Test 4: Express checkout tracking respects opt-out**
1. Add filter: `add_filter( 'wcpay_enable_shopper_tracking', '__return_false' );`
2. Load a page with Apple Pay / Google Pay buttons
3. Click the buttons
4. Verify NO tracking requests are made (previously these were hardcoded)

**Test 5: Backward compatibility**
1. Verify old `platform_tracks` endpoint still works
2. Test with and without batching
3. Ensure no JS errors in console

**Test 6: Unit tests**
```bash
npm run test:php -- --filter=WooPay_Tracker_Test
```

**Test 7: High-traffic simulation**
1. Rapidly click around site (product → cart → checkout)
2. Verify events are batched appropriately
3. Check server load is reduced vs individual requests

**Expected Results:**
- 70-85% reduction in tracking-related Ajax requests
- 5-10 requests reduced to 1-2 batched requests per session
- All events still accurately recorded with correct timestamps
- No impact on user experience

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _TBD_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by rtio** [comment]
Closing since this is not working as I expected.
